src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -754,6 +757,8 @@ public void runMayThrow() throws InterruptedException
                 ScheduledExecutors.nonPeriodicTasks.shutdown();
                 if (!ScheduledExecutors.nonPeriodicTasks.awaitTermination(1, MINUTES))
                     logger.warn("Miscellaneous task executor still busy after one minute; proceeding with shutdown");
+
+                logger.info("DrainOnShutdown completed in {} ms", SafeArg.of("ms", watch.elapsed(TimeUnit.MILLISECONDS)));
```

Comment:
Would it not be simpler to just add metrics? Or is there any advantages to using the logs?

Reply:
Can you add logs at the start of each function (so we understand if a SIGKILL was sent before the thread completed) flush each of the logs?

Reply:
added a log to drainOnShutdown

the `setMode` calls i.e:`setMode(Mode.DRAINING, "starting drain process", true);` will emit a log

```
    void setMode(Mode m, @Safe String msg, boolean log)
    {
        operationMode = m;
        if (log)
            logger.info(m.toString(), SafeArg.of("msg", msg));
        else
            logger.debug(m.toString(), SafeArg.of("msg", msg));
    }
```